whylog/assistant/regex_assistant/regex.py

```diff
@@ -3,13 +3,74 @@
 """
 
 import re
+from collections import deque
 
 from whylog.assistant.regex_assistant.exceptions import NotMatchingRegexError
+from whylog.assistant.span import Span
+from whylog.assistant.span_list import SpanList
 
 special_characters = frozenset('.^$*+{}?[]|()]')
 group_pattern = re.compile(r"[a-zA-Z]+|[0-9]+|\s+|[\W]+")
 
 
+def group_spans_from_regex(regex, text):
+    """
+    Creates spans from regex groups in regex corresponding to text.
+
+    Assumption: regex matches text.
+    """
+
+    if len(regex) < 2:
+        return []
+
+    verify_regex(regex, text)
+
+    matcher = re.match(re.compile(regex), text)
+    groups_count = len(matcher.groups())
+    group_ranges_in_text = [matcher.span(group_no + 1) for group_no in range(groups_count)]
+    group_ranges_in_text = [(start, end) for start, end in group_ranges_in_text]
+
+    group_ranges_in_regex = []
+    parenthesis_stack = deque()
+    for pos in range(1, len(regex)):
+        if not regex[pos - 1] == '\\':
```

Comment:
This condition is always true. You compare char with two elements consts string.